include/boost/histogram/detail/nstore.hpp

```diff
@@ -64,13 +69,13 @@ class nstore {
 
   // move semantics
   nstore(BOOST_RV_REF(nstore) o) :
-    size_(0),
-    depth_(d1),
-    buffer_(0)
+    size_(o.size_),
+    depth_(o.depth_),
+    buffer_(o.buffer_)
   {
-    std::swap(size_, o.size_);
-    std::swap(depth_, o.depth_);
-    std::swap(buffer_, o.buffer_);
+	o.depth_ = d1;
+	o.size_ = 0;
+	o.buffer_ = static_cast<void*>(0);
```

Comment:
Seems to me like it's done ;). It's actually a very minor thing (since the compiler would optimize that anyway), but you avoid a copy in std::swap.